### PR TITLE
Generate translation files on master changes

### PR DESCRIPTION
**Pull request by EvanHerman**
Generate CoBlocks translation files in `i18n` job on CircleCI and upload them to the artifacts for the build when any changes to `master` are detected. This includes any PRs merged into `master`, or any hot fixes pushed directly to `master`.

https://circleci.com/gh/godaddy-wordpress/coblocks/6319#artifacts/containers/0

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -124,6 +128,42 @@ jobs:
...
+          command: sudo apt-get install gettext
+      - run:
+          name: Install node packages
+          command: npm install
+      - run:
+          name: Build the plugin
+          command: |
+            mkdir -p /tmp/artifacts
+            npm run babel:build
+            npm run pot2json
```
Comment (fjarrett) [inline .circleci/config.yml]: @EvanHerman These lines can be removed. The `babel:build` script already does `makepot` which calls `pot2json`, and the other scripts are for dev use only. Basically all this job has to do is ensure two files are updated with every `master` commit: `coblocks.pot` and `coblocks.json`.

File: .circleci/config.yml [hunk 2]
```diff
@@ -124,6 +128,42 @@ jobs:
...
+            npm run json2po
+            npm run po2mo
+            npm run po2jed
+            grunt compress:coblocks
+            mv languages/coblocks-*.json /tmp/artifacts
+            mv coblocks-translations.zip /tmp/artifacts
```
Comment (fjarrett) [inline .circleci/config.yml]: @EvanHerman what is the `coblocks-translations.zip` actually for?

File: .circleci/config.yml (hunk 1 above)
Comment (fjarrett) [reply .circleci/config.yml]: Lines 156-159 can be removed

File: .circleci/config.yml [hunk 3]
```diff
@@ -124,6 +128,42 @@ jobs:
...
+          command: npm install
+      - run:
+          name: Build the plugin
+          command: |
+            mkdir -p /tmp/artifacts
+            npm run babel:build
+            npm run pot2json
+            npm run json2po
+            npm run po2mo
+            npm run po2jed
```
Comment (fjarrett) [inline .circleci/config.yml]: @EvanHerman These lines can be removed. The babel:build script already does makepot which calls pot2json, and the other scripts are for dev use only. Basically all this job has to do is ensure two files are updated with every master commit: coblocks.pot and coblocks.json.

**Comment by fjarrett** [comment]
@EvanHerman I think what we really want here is for the `coblocks.pot` and `coblocks.json` files to be regenerated and just committed back to the `master` branch, no artifacts needed

File: .circleci/config.yml (hunk 2 above)
Comment (EvanHerman) [reply .circleci/config.yml]: Sorry, I misunderstood. I thought we wanted to generate translation files for the translation team to test with on each commit. PR has been updated to reflect the actual needed changes.

**Pull request merged**